tests/drivers/counter/counter_basic_api/boards/stm32f4_disco.conf

```diff
@@ -0,0 +1 @@
+CONFIG_COUNTER_TIMER_STM32=y
```

Comment:
We should not have to enable this SoC specific flag manually.
We miss a way to get `COUNTER_TIMER_STM32` selected when COUNTER is selected and a counter node is enabled.

Please check how it is done for instance for `MEMC_STM32`:
In `drivers/memc/Kconfig.stm32`, it is defined as:
```
DT_COMPAT_ST_STM32_FMC := st,stm32-fmc

config MEMC_STM32
	bool "Enable STM32 Flexible Memory Controller (FMC)"
	default $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_FMC))
```
=> It is selected when a node with matching compatible is enabled.

This way, driver get enabled when and only when both following condtions are true:
- MEMC is enabled
- matching device tree node is enabled

Reply:
Oh okay, yeah this is better. Done.